lib/delete-users-using-local-file.rb

```diff
@@ -0,0 +1,109 @@
+require 'zendesk_api'
+require 'rest-client'
+require 'json'
+require 'date'
+
+require_relative 'zendesk-setup.rb'
+
+# Setup vars
+
+diag = "true"
+url = "#{ENV['ZENDESK_URL']}/deleted_users/"
+lastyear = Date.today.next_day - 365
+
+File.open "data/soft_deleted_users.json" do |input_file|
+
+  file_data = JSON.load input_file
+
+  file_data.each do |user|
+
+
+    # Extract some fields
+    updated_at = user["updated_at"]
+    last_login_at = user["last_login_at"]
+    user_id = user["id"]
+    active = user["active"]
+    name = user["name"]
+
+    # Convert strings to dates, if strings are nil then set a false date of 2012 so comparison works.
+    # Booleans returned by API are not true boolean but strings.
+
+    if active == "false"
+      # account is already soft deleted so force hard delete
+      hard_delete = "true"
+    end
+
+    false_date = "2012-01-01"
+
+    if hard_delete != "true"
+      if updated_at.nil?
+        updated_at = false_date
+      end
+
+      # parse the date so we can do comparisons
+
+      updated = Date.parse(updated_at)
+
+      if last_login_at != nil
+        last_login = Date.parse(last_login_at)
+      else
+        last_login_at = false_date
+        last_login = Date.parse(last_login_at)
+      end
+
+      # diagnostics 
+
+      if diag == "true"
+        puts "user_id: #{user_id}"
+        puts "Name: #{name}"
+        puts "Active: #{active}"
+        puts "last_login_at: #{last_login_at}"
+        puts "last_login: #{last_login}"
+        puts "url: #{url}"
+        puts "lastyear: #{lastyear}"
+      end
+
+    # If last logged in < last year, let's check whether user has any tickets associated
+
+      if last_login <= lastyear
+        puts "Potential DELETION candidate - check tickets"
+        count = @client.search!(:query => "requester_id:#{user_id}").count
+        ticket_count = Integer count
+        puts "ticket_count: #{ticket_count}"
+      else
+        puts "DO NOT DELETE"
+      end
+    else
+    ticket_count = 0
```

Comment:
Can we move `ticket_count = 0` from line 77 to line 28 (i.e. outside of if condition)? The variable is created within if condition and it will be lost when a program goes out of if condition.

Reply:
Resolved as suggested